include/up-client-vsomeip-cpp/routing/SomeipRouterInterface.hpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SOMEIP_ROUTER_INTERFACE_HPP
+#define SOMEIP_ROUTER_INTERFACE_HPP
+#include <up-cpp/transport/UTransport.h>
+#include <up-client-vsomeip-cpp/translation/MessageTranslator.hpp>
+#include <memory>
+
+class SomeipHandler;
+
+/**
+ * @brief Represents an interface for the SomeIP router.
```

Comment:
Do you expect to have other implementations? Is this to be able to map to sommR vs vsomeip? I don't think we need this anymore as the library will be written for a specific implementation.